AdaptorsDataAccessPlugins/src/au/gov/asd/tac/constellations/dataaccess/adaptors/providers/gaffer/GafferSimpleQuery.java

```diff
@@ -0,0 +1,128 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package au.gov.asd.tac.constellations.dataaccess.adaptors.providers.gaffer;
+
+import au.gov.asd.tac.constellation.graph.processing.GraphRecordStoreUtilities;
+import au.gov.asd.tac.constellation.graph.processing.RecordStore;
+import au.gov.asd.tac.constellation.graph.schema.visual.concept.VisualConcept;
+
+import java.io.IOException;
+import java.util.List;
+import org.openide.util.Exceptions;
+
+import uk.gov.gchq.gaffer.commonutil.iterable.CloseableIterable;
+import uk.gov.gchq.gaffer.data.element.Edge;
+import uk.gov.gchq.gaffer.data.element.Element;
+import uk.gov.gchq.gaffer.data.element.Entity;
+import uk.gov.gchq.gaffer.operation.OperationChain;
+import uk.gov.gchq.gaffer.operation.impl.get.GetAdjacentIds;
+import uk.gov.gchq.gaffer.operation.impl.get.GetElements;
+
+/**
+ *
+ * @author GCHQDeveloper601
+ */
+public class GafferSimpleQuery {
+
+    private String url;
+    private GafferConnector connector ;
+
+    public GafferSimpleQuery (){
+    //NOOP
+    }
+    
+    public void setUrl(String url){
+        this.url = url;
+        connector=new GafferConnector(url);
+    }
+    
+    protected void setGafferConnectorService(GafferConnector gafferConnector){
+        connector=gafferConnector;
+    }
+    
+    /**
+     * This function will only return the details of the Elements being queries
+     * without getting any hops
+     *
+     * @param queryIds The name value of {@link uk.gov.gchq.gaffer.data.element.Entity)
+     * @param recordStore The record store to load the results into
+     */
+    public void queryForDetails(List<String> queryIds, RecordStore recordStore) {
+        //This query is not quite right yet so is not yet enabled.
+        GetElements elms = new GetElements.Builder().input(queryIds).build();
+        OperationChain<CloseableIterable<? extends Element>> opChain = new OperationChain.Builder().first(elms).build();
+        fetchResults(opChain, recordStore);
+    }
+
+    /**
+     * This function will return 1 hop from the seed being queried
+     *
+     * @param queryIds The name value of {@link uk.gov.gchq.gaffer.data.element.Entity)
+     * @param recordStore The record store to load the results into
+     */
+    public void queryForOneHop(List<String> queryIds, RecordStore recordStore) { 
+        var opChain =buildOneHopChain(queryIds);
+        fetchResults(opChain, recordStore);
+    }
+
+    /**
+     * This function will return 2 hop from the seed being queried
+     *
+     * @param queryIds The name value of {@link uk.gov.gchq.gaffer.data.element.Entity)
+     * @param recordStore The record store to load the results into
+     */
+    public void queryForTwoHop(List<String> queryIds, RecordStore recordStore) {
+        var opChain = buildTwoHopChain(queryIds);
```

Comment:
final var - and it's probably personal preference but i'd type this as it's not immediately obvious what type is returned.